Windows with AVX512F is failing a typing test

From the numpy-wheels repo, see [this comment](https://github.com/MacPython/numpy-wheels/pull/104#issuecomment-779355607). The windows runs have been failing [since Feb 14](https://dev.azure.com/numpy/numpy/_build?definitionId=8&_a=summary). The failing test is in static typing:
```
with pytest.raises(np.ComplexWarning):
>       np.einsum("i,i->i", AR_LIKE_U, AR_LIKE_U, dtype=float, casting="unsafe", out=OUT_c)
E       Failed: DID NOT RAISE <class 'numpy.ComplexWarning'>
```

and occurs with an Azure machine that reports
```
NumPy CPU features: SSE SSE2 SSE3 SSSE3* SSE41* POPCNT* SSE42* AVX* F16C* \
    FMA3* AVX2* AVX512F* AVX512CD* AVX512_SKX* AVX512_CLX? AVX512_CNL?
```

where the machines in the regular CI report
```
NumPy CPU features: SSE SSE2 SSE3 SSSE3* SSE41* POPCNT* SSE42* AVX* F16C* \
    FMA3* AVX2* AVX512F? AVX512CD? AVX512_SKX? AVX512_CLX? AVX512_CNL?
```

Note the `AVX512F`, `AVX512CD`, `AVX512_SKX` difference.

@seiko2plus, @BvB93 ideas? Is there a way to reproduce this with concrete types?

I am not sure what the difference is between the CI configuration and why they are getting different machines. A successful run log is [here](https://dev.azure.com/numpy/numpy/_build/results?buildId=15746&view=logs&j=c499766f-449b-5541-ed00-14b3885196ca), a failure is [here](https://dev.azure.com/numpy/numpy/_build/results?buildId=15517&view=logs&j=7031352c-8337-5b08-f106-f557e28887f8)

--- a/numpy/typing/tests/data/pass/einsumfunc.py
+++ b/numpy/typing/tests/data/pass/einsumfunc.py
@@ -2,7 +2,6 @@ from __future__ import annotations
 
 from typing import List, Any
 
-import pytest
 import numpy as np
 
 AR_LIKE_b = [True, True, True]
@@ -12,6 +11,7 @@ AR_LIKE_f = [1.0, 2.0, 3.0]
 AR_LIKE_c = [1j, 2j, 3j]
 AR_LIKE_U = ["1", "2", "3"]
 
+OUT_f: np.ndarray[Any, np.dtype[np.float64]] = np.empty(3, dtype=np.float64)
 OUT_c: np.ndarray[Any, np.dtype[np.complex128]] = np.empty(3, dtype=np.complex128)
 
 np.einsum("i,i->i", AR_LIKE_b, AR_LIKE_b)
@@ -25,8 +25,7 @@ np.einsum("i,i,i,i->i", AR_LIKE_b, AR_LIKE_u, AR_LIKE_i, AR_LIKE_c)
 np.einsum("i,i->i", AR_LIKE_f, AR_LIKE_f, dtype="c16")
 np.einsum("i,i->i", AR_LIKE_U, AR_LIKE_U, dtype=bool, casting="unsafe")
 np.einsum("i,i->i", AR_LIKE_f, AR_LIKE_f, out=OUT_c)
-with pytest.raises(np.ComplexWarning):
-    np.einsum("i,i->i", AR_LIKE_U, AR_LIKE_U, dtype=float, casting="unsafe", out=OUT_c)
+np.einsum("i,i->i", AR_LIKE_U, AR_LIKE_U, dtype=int, casting="unsafe", out=OUT_f)
 
 np.einsum_path("i,i->i", AR_LIKE_b, AR_LIKE_b)
 np.einsum_path("i,i->i", AR_LIKE_u, AR_LIKE_u)

